src/types/serialize.rs

```diff
@@ -36,7 +34,7 @@ impl From<Type> for SerSimpleType {
             TypeEnum::Prim(t) => match t {
                 PrimType::Extension(c) => SerSimpleType::Opaque(c),
                 PrimType::Alias(a) => SerSimpleType::Alias(a),
-                PrimType::Function(sig) => SerSimpleType::G(Box::new(*sig)),
+                PrimType::Function(sig) => SerSimpleType::G(sig),
```

Comment:
Hmmmm, reckon we could have done this one-line change without this PR, but I think it's a good drive-by